### ISSUE TITLE
Consider adding System.ComponentModel.DataAnnotations.DisplayAttribute

### ISSUE THREAD

**Issue by DanTup** [bridgedotnet/Bridge#2559]
[This class](https://msdn.microsoft.com/en-us/library/system.componentmodel.dataannotations.displayattribute(v=vs.110).aspx) is quite useful for adding some extra metadata to things; for example if you want to reflect over an enum and get friendlier names:

```cs
public enum MyStatus {
    Pending,
    Approved,
    [Display(Name = "Awaiting Cancellation")]
    AwaitingCancellation
}
```

Currently in order to reuse our C# enums on the client side we use "linked files" but then have to provide a stub version of this class in order to compile.

It looks like this attribute is fairly isolated and is quite useful; would you consider adding it?

**Comment by geoffreymcgill** [comment]
Agreed. This would be very useful and seems like it would be relatively easy to implement. 

We will discuss.

**Comment by michaelcheers** [comment]
@DanTup, can I just add it as NonScriptable and External to #2787 or do you want more functionality than just being ignored?

**Comment by DanTup** [comment]
I think it just needs to exist with the properties (like Name) and be output as normal. If you make it nonscriptable/external then presumably we wouldn't be able to use reflection to read the `Name` property (which is the goal).

**Comment by michaelcheers** [comment]
@DanTup, I think mostly this attribute could replace the Name attribute interestingly, though it could be nice to have a js name and a reflection name.

**Comment by DanTup** [comment]
I think the purposes are very different. The Bridge Name attribute is for controlling output of javascript names; this one is just for attaching metadata to your own classes. I think the purposes are very different and they should be kept separate (it wouldn't make sense for Bridge to change your JS output based on this attribute, and the purpose of this case is to have a shared attribute in C# + Bridge, so we can't use the Bridge one for what we need).
